File: Resueltos_por_competidor/Tomás_Triana_Galvis/Bigrams.cpp
#include<bits/stdc++.h>
typedef long long ll;
using namespace std;
/*
 * Autor: Tomás Triana Galvis
 * Problema: Bigrams
 * Juez online: Codeforces 2242A
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/2242/A
 **/ 
int main(){
    ll t,n,cuenta2,a;
    bool ans;
    cin>>t;
    while(t--){
        ans = false;
        cuenta2=0;
        cin>>n;
        for(int i=0;i<n;i++){
            cin>>a;
            if(a>=3){
                ans =true;
            }
            if(a==2){
                cuenta2++;
            }
        }
        if(cuenta2>=2){
            ans =true;
        }
        if(ans){
            cout<<"YES\n";
        }else{
            cout<<"NO\n";
        }
    }
    return 0;
}

File: Resueltos_por_competidor/Santiago_Ortiz/Bigrams.java
import java.util.Scanner;

/*
 * Autor: Darkovski7   
 * Problema: Bigrams
 * Juez Online: Codeforces
 * Veredicto: Accepted 
 * Url: https://codeforces.com/problemset/problem/2242/A
 */

public class Bigrams {

	static int contador1;
	static int contador2;

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		int t;
		t = sc.nextInt();

		for (int i = 0; i < t; i++) {
			int k;
			k = sc.nextInt();
			int[] x = new int[k];

			contador1 = 0;
			contador2 = 0;
			
			for (int j = 0; j < k; j++) {
				x[j] = sc.nextInt();

				if (x[j] >= 3) {
					contador1++;
				}
				if (x[j] >= 2) {
					contador2++;
				}
			}
			if (contador1 >= 1) {
				System.out.println("YES");
			}
			else if (contador2 >= 2) {
				System.out.println("YES");
			} else {
				System.out.println("NO");
			}

		}
	}

}
